six/modules/c++/six.sicd/include/six/sicd/SICDMesh.h

```diff
@@ -0,0 +1,202 @@
+    const std::vector<double>& getY() const
+    {
+        return mY;
+    }
+
+    //! \return The vector of Field descriptors (file name, type)
+    virtual std::vector<Mesh::Field> getFields() const;
+
+    /*!
+     * Serializes the mesh to binary
+     * \param[out] values The serialized data.
+     */
+    virtual void serialize(std::vector<sys::byte>& values) const;
+
+    /*!
+     * Deserializes from binary to a mesh. 
+     * \param values Data to deserialize.
+     */
+    virtual void deserialize(sys::byte*& values);
+
+protected:
+    const bool mSwapBytes;
+    std::string mName;
+    types::RowCol<size_t> mMeshDims;
+    std::vector<double> mX;
+    std::vector<double> mY;
+};
+
+/*!
+ *  \class NoiseMesh
+ *  \brief A storage structure that holds position and noise
+ *   information. A noise mesh holds position information and
+ *   noise-related values at these positions.
+ */
+class NoiseMesh : public PlanarCoordinateMesh
+{
+public:
+
+    /*!
+     * NoiseMesh constructor. Mesh data is uninitialized.
+     * \param name Name of the mesh.
+     */
+    NoiseMesh(const std::string& name);
+
+    /*!
+     * NoiseMesh constructor. Mesh data is uninitialized.
+     * \param name Name of the mesh.
+     * \param meshDims The mesh dimensions.
+     * \param x The x-coordinates for the mesh.
+     * \param y The y-coordinates for the mesh.
+     * \param mainBeamNoise Main beam noise values.
+     * \param azimuthAmbiguityNoise Azimuth ambiguity noise values.
+     * \param combinedNoise Combined noise values.
+     */
+    NoiseMesh(const std::string& name,
+              const types::RowCol<size_t>& meshDims,
+              const std::vector<double>& x,
+              const std::vector<double>& y,
+              const std::vector<double>& mainBeamNoise,
+              const std::vector<double>& azimuthAmbiguityNoise,
+              const std::vector<double>& combinedNoise);
+
+    //! \return The flat main beam noise over the mesh (amp)
+    const std::vector<double>& getMainBeamNoise() const
+    {
+        return mMainBeamNoise;
+    }
+
+    //! \return The flat azimuth ambiguity noise over the mesh (amp)
+    const std::vector<double>& getAzimuthAmbiguityNoise() const
+    {
+        return mAzimuthAmbiguityNoise;
+    }
+
+    //! \return The flat noise floor values over the mesh (power)
+    const std::vector<double>& getCombinedNoise() const
+    {
+        return mCombinedNoise;
+    }
+
+    /*!
+     * Get the mesh fields and types.
+     * \return The vector of Field descriptions
+     */
+    std::vector<Mesh::Field> getFields() const;
+
+    /*!
+     * Serializes the mesh to binary
+     * \param[out] values The serialized data.
+     */
+    virtual void serialize(std::vector<sys::byte>& values) const;
+
+    /*!
+     * Deserializes from binary to a mesh. 
+     * \param values Data to deserialize.
+     * \param swapBytes Should byte-swapping be applied?
+     */
+    virtual void deserialize(sys::byte*& values);
```

Comment:
Why is this passed in as a reference to a pointer?  So that you can chain a bunch of objects like this together?  The behavior should be really clearly documented in the base class and at least touched on here.

Also, should be able to pass this in everywhere by `const sys::byte*&` right?

Reply:
It's now const, I'm not sure what I can add to the documentation to make this more clear.